### PR TITLE
Create forgetchild-call-super

### PR DESCRIPTION
**Pull request by chunhtai**

### CODE REVIEW

File: src/docs/release/breaking-changes/forgetchild-call-super [hunk 1]
```diff
@@ -0,0 +1,84 @@
+---
+title: Element subclasses must call super in forgetChild
+description: Any element subclasses that override forgetChild will require to call super.
+---
+
+## Summary
+
+A recent global key duplication detection refactor requires `Element` subclasses which override
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/forgetchild-call-super]: which => that

File: src/docs/release/breaking-changes/forgetchild-call-super [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+`forgetChild` to call super.
+
+## Context
+
+When we encounter a global key duplication which will be cleaned up by an element rebuild later, we
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/forgetchild-call-super]: Can you rewrite this section to be past tense? By the time developers look at it, it's no longer the "current" implementation. It's the previous implementation.

File: src/docs/release/breaking-changes/forgetchild-call-super [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+build and only verify global key duplication at the end of the build instead of throwing error
+immediately. In order to correctly differentiate between a real global key duplication and the case
+we mentioned earlier, we would implement a mechanism to remove previous global key duplication in
+forgetChild if the rebuild has happened. This will, however, require all Element subclasses that
+override forgetChild to call the super method.
+
+## Description of change
+
+The `forgetChild` of abstract class `Element` has a base implementation to remove global key reservation,
+and is enforced by ``@mustCallSuper` meta tag. The subclasses will have to call super; otherwise,
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/forgetchild-call-super]: The double back tick is an error? (before @mustCallSuper). Rewrite this in past tense, too.

File: src/docs/release/breaking-changes/forgetchild-call-super [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+analyzer will show linting error and global key duplication detection might throw unexpected error.
+
+## Migration guide
+
+For example, an app has a `Element` subclass override `forgetChild`
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/forgetchild-call-super]: => In the following example, an app's `Element` subclass overrode the `forgetChild` method.

File: src/docs/release/breaking-changes/forgetchild-call-super [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    void forgetChild(Element child) {
+        ...
+        super.forgetChild(child);
+    }
+}
+```
+
+## Timeline
+
+{% comment %}
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/forgetchild-call-super]: Which build this is in?

Comment (chunhtai) [reply src/docs/release/breaking-changes/forgetchild-call-super]: The change is not landed yet, I am not sure which build will this be in.
I am not entirely sure about this process. I am trying to follow https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

Should I only open up this pr after the change has been landed?

**Comment by chunhtai** [comment]
> Besides the issues I've indicated, please add this change to the index file.

Thanks for reviewing ! will update them

File: src/docs/release/breaking-changes/forgetchild-call-super (hunk 5 above)
Comment (sfshaza2) [reply src/docs/release/breaking-changes/forgetchild-call-super]: I've started an email thread with @Hixie, so stay tuned.

**Comment by sfshaza2** [comment]
@chunhtai, Hixie agreed that we shouldn't add these pages to the website until the change has landed. He has updated the instructions.

**Comment by chunhtai** [comment]
@sfshaza2 thanks, i will reopen this once it is landed

**Comment by chunhtai** [comment]
updated the release version, this pr is ready for review

File: src/docs/release/breaking-changes/forgetchild-call-super (hunk 5 above)
Comment (sfshaza2) [reply src/docs/release/breaking-changes/forgetchild-call-super]: Following up. Yes, open a PR once the breaking change has landed.

**Comment by chunhtai** [comment]
bump for review

**Pull request merged**